src/terminal/RenderBufferBuilder.cpp

```diff
@@ -449,6 +449,7 @@ ColumnCount RenderBufferBuilder<Cell>::renderUtf8Text(CellLocation screenPositio
     auto graphemeClusterSegmenter = unicode::utf8_grapheme_segmenter(text);
     for (u32string const& graphemeCluster: graphemeClusterSegmenter)
     {
+        screenPosition.column = ColumnOffset::cast_from(columnCountRendered);
```

Comment:
I think this won't work if `screenPosition.colum` - as passed by the caller - is a non-zero value.
This code assumes that `screenPosition.column` would always be zero and then offsetting into it by columnCountRendered works. It wont however when `screenPosition.column` is greater than `0`.

This won't be generally the case for standard text, but (I'm not sure), it may very well be that for non-terminal text, when this function is used, the screen position might not always be forced to start at the left-most column (`0`).

NB: This function is reused for the host-programmable and indicator statusline.